### PR TITLE
Events: Implement common event detection interface

### PR DESCRIPTION
**Pull request by dkrako**
All event detection methods now have both positions and velocities parameters, independent of their actual usage.

This change also includes adding a new check that can be used across modules.

This change is the prerequisite for two things:
* a more straight-forward implementation in the coming `detect_events()` method in the `Dataset` class
* a greater range of properties to be included in the event dataframe

### CODE REVIEW

File: src/pymovements/events/engbert.py [hunk 1]
```diff
@@ -73,6 +77,12 @@ def microsaccades(
         If `threshold` value is below `min_threshold` value.
         If passed `threshold` is either not two-dimensional or not a supported method.
     """
+    positions = np.array(positions)
+    velocities = np.array(velocities)
+
+    # Check correct shapes of positions and velocities.
```
Comment (SiQube) [inline src/pymovements/events/engbert.py:83]: I don't think this comment is necessary

File: src/pymovements/events/idt.py [hunk 2]
```diff
@@ -84,24 +89,19 @@ def idt(
...
+    velocities = np.array(velocities)
 
-    # make sure positions is 2d
-    if positions.ndim != 2:
-        raise ValueError('positions needs to have shape (N, 2)')
-
-    # make sure positions has shape (N, 2)
-    if positions.shape[1] != 2:
-        raise ValueError('positions needs to have shape (N, 2)')
+    # Check correct shapes of positions and velocities.
```
Comment (SiQube) [inline src/pymovements/events/idt.py:95]: I think this comment is unnecessary

File: src/pymovements/events/idt.py [hunk 3]
```diff
@@ -84,24 +89,19 @@ def idt(
...
+    check_shapes_positions_velocities(positions=positions, velocities=velocities)
 
     # Check if dispersion_threshold is greater 0
     if dispersion_threshold <= 0:
         raise ValueError('dispersion threshold must be greater than 0')
 
-    # Check if duration_threshold is greater 0
+    # Check if minimum_duration is greater 0
```
Comment (SiQube) [inline src/pymovements/events/idt.py:102]: unnecessary comment

File: src/pymovements/events/ivt.py [hunk 4]
```diff
@@ -74,29 +75,12 @@ def ivt(
...
-    if velocities.shape[1] != 2:
-        raise ValueError('velocities need to have shape (N, 2)')
-
-    # Check matching shape for positions and velocities
-    if positions.shape != velocities.shape:
-        raise ValueError(
-            f"shape of positions {positions.shape} doesn't match"
-            f'shape of velocities {velocities.shape}',
-        )
+    # Check correct shapes of positions and velocities.
```
Comment (SiQube) [inline src/pymovements/events/ivt.py:78]: delete?

File: src/pymovements/events/engbert.py [hunk 5]
```diff
@@ -28,10 +28,12 @@
 import polars as pl
 
 from pymovements.transforms import consecutive
+from pymovements.utils.checks import check_shapes_positions_velocities
 
 
 def microsaccades(
-    velocities: np.ndarray,
+    positions: list[list[float]] | np.ndarray,
+    velocities: list[list[float]] | np.ndarray,
```
Comment (SiQube) [inline src/pymovements/events/engbert.py:36]: should we here do list[list[float]] or list[tuple[float]]?

File: src/pymovements/events/engbert.py [hunk 6]
```diff
@@ -48,6 +50,8 @@ def microsaccades(
 
     Parameters
     ----------
+    positions : np.ndarray, shape (N, 2)
+        x and y positions of N samples in chronological order
```
Comment (SiQube) [inline src/pymovements/events/engbert.py:54]: so we only target one eye?

**Review by SiQube — APPROVED** [review]

File: src/pymovements/events/engbert.py (hunk 6 above)
Comment (dkrako) [reply src/pymovements/events/engbert.py:54]: I think we had this discussion already a few times, but lets keep it in #28 in the future, where we can decide on how to support binocular event detection one day.

just a basic summary of our past discussion outcomes:
* all our implemented event detection algorithms (and also the ones which we plan to implement) are only defined for a single eye
* we want to implement this functionality in a binocular filter in the preprocessing pipeline #28
* there we can also support different kinds of methods for merging the two eyes
* until then, the event detection will be purely monocular

**Pull request merged**